hack/kubelet-plugin-prestart.sh

```diff
@@ -0,0 +1,174 @@
+if [ "${NVIDIA_DRIVER_ROOT}" == "/run/nvidia/driver" ]; then
+    # Expectation: link may be broken initially if the GPU operator isn't
+    # deployed yet. The link heals once GPU operator provides the driver on the
+    # host at /run/nvidia/driver. Notably, on the host, the directory
+    # /run/nvidia is the mount point that gets created when the GPU operator
+    # gets deployed.
+    echo "create symlink: /driver-root -> /host-run/nvidia/driver"
+    ln -s /host-run/nvidia/driver /driver-root
+    # stat /driver-root
+else
+    echo "create symlink: /driver-root -> /host-driver-root"
+    ln -s /host-driver-root /driver-root
+fi
+
+emit_common_err () {
+    printf '%b' \
+        "Check failed. Has the NVIDIA GPU driver been set up? " \
+        "It is expected to be installed under " \
+        "NVIDIA_DRIVER_ROOT (currently set to '${NVIDIA_DRIVER_ROOT}') " \
+        "in the host filesystem. If that path appears to be unexpected: " \
+        "review the DRA driver's 'nvidiaDriverRoot' Helm chart variable. " \
+        "Otherwise, review if the GPU driver has " \
+        "actually been installed under that path.\n"
+}
+
+validate_and_exit_on_success () {
+    echo -n "$(date -u +"%Y-%m-%dT%H:%M:%SZ")  /driver-root (${NVIDIA_DRIVER_ROOT} on host): "
+
+    # Search specific set of directories (not recursively: not required, and
+    # /driver-root may be a big tree). Limit to first result (multiple results
+    # are a bit of a pathological state, but continue with validation logic).
+    # Suppress find stderr: some search directories are expected to be "not
+    # found".
+
+    NV_PATH=$( \
+        find \
+            /driver-root/bin \
+            /driver-root/sbin \
+            /driver-root/usr/bin \
+            /driver-root/sbin \
+        -maxdepth 1 -type f -name "nvidia-smi" 2> /dev/null | head -n1
+    )
+
+    # Follow symlinks (-L), because `libnvidia-ml.so.1` is typically a link.
+    # maxdepth 1 also protects against any potential symlink loop (we're
+    # suppressing find's stderr, so we'd never see messages like 'Too many
+    # levels of symbolic links').
+    NV_LIB_PATH=$( \
+        find -L \
+            /driver-root/usr/lib64 \
+            /driver-root/usr/lib/x86_64-linux-gnu \
+            /driver-root/usr/lib/aarch64-linux-gnu \
+            /driver-root/lib64 \
+            /driver-root/lib/x86_64-linux-gnu \
+            /driver-root/lib/aarch64-linux-gnu \
+        -maxdepth 1 -type f -name "libnvidia-ml.so.1" 2> /dev/null | head -n1
+    )
+
+    if [ -z "${NV_PATH}" ]; then
+        echo -n "nvidia-smi: not found, "
+    else
+        echo -n "nvidia-smi: '${NV_PATH}', "
+    fi
+
+    if [ -z "${NV_LIB_PATH}" ]; then
+        echo -n "libnvidia-ml.so.1: not found, "
+    else
+        echo -n "libnvidia-ml.so.1: '${NV_LIB_PATH}', "
+    fi
+
+    # Log top-level entries in /driver-root (this may be valuable debug info).
+    echo "current contents: [$(/bin/ls -1xAw0 /driver-root 2>/dev/null)]."
```

Comment:
This shows what I was trying to achieve (condensed information on a single line, also showing how the contents of `/driver-root` evolved during init container runtime):

```
2025-06-08T14:13:52Z  /driver-root (/run/nvidia/driver on host): nvidia-smi: not found, libnvidia-ml.so.1: not found, current contents: [].
2025-06-08T14:14:02Z  /driver-root (/run/nvidia/driver on host): nvidia-smi: not found, libnvidia-ml.so.1: not found, current contents: [lib].
```